django/contrib/postgres/fields/array.py

```diff
@@ -219,6 +220,11 @@ def as_sql(self, qn, connection):
         sql = '%s::%s' % (sql, self.lhs.output_field.db_type(connection))
         return sql, params
 
+    def process_rhs(self, compiler, connection):
+        if isinstance(self.rhs, Query):
+            return 'ARRAY(%s)' % self.rhs, []
```

Comment:
You shouldn't use the string representation of the query but someting along the lines of what `django.db.models.fields.related.related_lookups.RelatedIn` does.